chain-signatures/node/src/cli.rs

```diff
@@ -161,6 +161,35 @@ impl Cli {
     }
 }
 
+fn spinup_indexer(
+    options: &indexer::Options,
+    mpc_contract_id: &AccountId,
+    account_id: &AccountId,
+    sign_queue: &Arc<RwLock<SignQueue>>,
+    gcp: &GcpService,
+) -> std::thread::JoinHandle<()> {
+    let options = options.clone();
+    let mpc_contract_id = mpc_contract_id.clone();
+    let account_id = account_id.clone();
+    let sign_queue = sign_queue.clone();
+    let gcp = gcp.clone();
+    std::thread::spawn(move || {
+        // If indexer fails for whatever reason, let's spin it back up with exponential delay:
+        for i in 0..10 {
+            let options = options.clone();
+            let mpc_contract_id = mpc_contract_id.clone();
+            let account_id = account_id.clone();
+            let sign_queue = sign_queue.clone();
+            let gcp = gcp.clone();
+
+            if let Err(err) = indexer::run(options, mpc_contract_id, account_id, sign_queue, gcp) {
```

Comment:
I guess we need to exit the loop if the ::run() is successful?

Reply:
so, there's no cancellation when it comes to indexer, since it's ran on a separate thread. So that `indexer::run` will never return anything unless it errors out. The reason why none of our stuff stalls is because the integration tests will directly kill the nodes. I'll add in a `break` to make it explicit just in case we do add interrupt handlers in the future, but this is not an issue at all for us